app/src/androidTest/java/de/westnordost/streetcomplete/data/logs/LogsDaoTest.kt

```diff
@@ -0,0 +1,78 @@
+package de.westnordost.streetcomplete.data.logs
+
+import de.westnordost.streetcomplete.data.ApplicationDbTestCase
+import de.westnordost.streetcomplete.data.logs.LogLevel.*
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class LogsDaoTest : ApplicationDbTestCase() {
+    private lateinit var dao: LogsDao
+
+    @BeforeTest fun createDao() {
+        dao = LogsDao(database)
+    }
+
+    @Test fun getAll() {
+        val m1 = createMessage("1", timestamp = 1)
+        val m2 = createMessage("2", timestamp = 100)
+        val m3 = createMessage("3", timestamp = 10)
+
+        listOf(m1, m2, m3).forEach { dao.add(it) }
+
+        // sorted by timestamp descending
+        assertEquals(listOf(m2, m3, m1), dao.getAll())
+    }
```

Comment:
How about this?

```suggestion
    @Test fun getAll_sorts_by_timestamp_descending() {
        val m1 = createMessage("1", timestamp = 1)
        val m2 = createMessage("2", timestamp = 100)
        val m3 = createMessage("3", timestamp = 10)

        listOf(m1, m2, m3).forEach { dao.add(it) }

        assertEquals(listOf(m2, m3, m1), dao.getAll())
    }
```
(etc.)

I think the tests are not written completely consistent, but most tests have as names what behavior they test